pyccel/codegen/printing/ccode.py

```diff
@@ -1033,15 +1036,21 @@ def _print_Constant(self, expr):
         return self._print(val)
 
     def _print_Return(self, expr):
-        code = ''
         args = [VariableAddress(a) if self.stored_in_c_pointer(a) else a for a in expr.expr]
-        if expr.stmt:
-            code += self._print(expr.stmt)+'\n'
+        if len(args) > 1:
+            if expr.stmt:
+                return self._print(expr.stmt)+'\n'+'return 0;'
+            return 'return 0;'
         if len(args) == 1:
-            code +='return {0};'.format(self._print(args[0]))
-        elif len(args) > 1:
-            code += 'return 0;'
-        return code
+            if isinstance(expr.stmt, CodeBlock):
+                for a in expr.stmt.body:
+                    if isinstance(a, Assign):
+                        b = a
+                if isinstance(b.rhs, IndexedElement):
+                    return '{0}\nreturn {1};'.format(self._print(expr.stmt), self._print(b.lhs))
```

Comment:
It is not all `IndexedElements` that are problematic, just the ones whose array needs deallocating. For example:
```
def f(a : 'int[:]'):
    return a[0]
```
Can be translated without temporary variables

Reply:
Another example:
```
@stack_array('b')
def f(a : int):
    b = np.ones(4)
    return b[a]
```
Here there is also no `Deallocate` statement